### ISSUE TITLE
New command line interface

### ISSUE THREAD

**Issue by pylover** [brython-dev/brython#1689]
`Brython` is awesome but it's command line interface not. some tasks(sub-commands) are combined with options for example `--modules`.

Just imagine:

```bash
brython modules --exclude PATTERN1 --exclude PATTERN1 path/to/search/dir
```

Toughts?

**Comment by schmave** [comment]
Can you say more about what you are hoping for? Removing the `--` from `modules` and other tasks/commands seems straightforward enough, though I'm not sure the benefit outweighs the annoyance of changing it for existing users. Adding an `--exclude` option to `--modules` could be helpful. Are there other things?

**Comment by pylover** [comment]
Sub-command stands for a task or routine, but options (aka optional arguments: `-f/--foo` mostly used to change the behaviour of a routine or task. for example, the `pip` command does more than one thing and these various features are encapsulated into sub-commands like `install [-f]`, `uninstall` and etc...

An option, let's talk about the `-f` in `pip install` is useful to change the Install sub-command behaviour.

It helps to easily extend and develop separate sub-commands without worrying about option conflict and a lot of pitfalls. 

I think it's no need for a major release, just introduce a new command-line interface as well as the existing one.

I suggest https://github.com/pylover/easycli for command-line interface multiplexing and https://github.com/pylover/bddcli for testing.

Please check out [this tutorial](https://pylover.github.io/easycli/tutorial.html) and inform me if interested. I can send a PR for a new command-line interface for Brython with 100% code coverage and responsibility.

**Comment by pylover** [comment]
Suggested sub-commands:

```bash
brython init                       # same as brython-cli --install
brython create-dist                # same as brython-cli --make_dist
brython optimise-stdlib  [--reset] # same as brython-cli --modules and --reset
brython serve [-p PORT]            # same as brython-cli --server
brython update-runtime             # same as brython-cli --update
```

**Comment by PierreQuentel** [comment]
`brython-cli` cannot be replaced by `brython` because on Windows, when you are in a folder with a file `brython.js` in it, typing `brython` in the command line triggers this error message:

![error](https://user-images.githubusercontent.com/3447215/121226415-d33d8b00-c88a-11eb-98d8-03c754b65f3e.png)

Apart from that, I don't really see how "init" is more appropriate than "install", "create-dist" than "make-dist" etc. I am not implying that the present CLI is perfect, but the syntax has been used for years, and there should be convincing arguments to change it.

**Comment by pylover** [comment]
Of course, whatever you say. 

I think the main problem is mixing optional, positional and sub-commands. please tell me about your prefered names for the sub-commands and arguments, there is no insist about naming at all. I'm just trying to explain my idea with names.

The current command-line interface is missing some options such as `--exclude`, destination and target filename or read/write to stdin/stdout. 

Anyway, extending the current command-line interface is not interested because it's hard to solve conflicts between verbs and options.

In short: the main idea is to separate verbs, options and positional arguments by implementing a new command-line interface for new users. perhaps a way to achieve an impeccable command-line interface for the Brython.

I have another option too: A new project that just uses Brython as a dependency to deliver a new shell interface.

In the end, thanks a lot in advance.

**Comment by pylover** [comment]
Just remember the Zen of Python:

```
Beautiful is better than ugly.
Explicit is better than implicit.
Simple is better than complex.
Complex is better than complicated.
Flat is better than nested.
Sparse is better than dense.
Readability counts.
Special cases aren't special enough to break the rules.
.
.
.
```

**Comment by pylover** [comment]
https://pypi.org/project/brythoncli/

**Comment by pylover** [comment]
Just an example of my makefile with the new cli:

```make
# WWW

WWW = www
WWWDIST = $(shell readlink -f $(WWW)/build)
DIAL = dial

$(WWWDIST)/dial.js:
	brython pack \
		--package-directory $(DIAL) \
		--output-directory $(WWWDIST) \
		dial
	
$(WWWDIST)/stdlib.min.js:
	- mkdir -p $(WWWDIST)
	brython pack-dependencies \
		--output-directory $(WWWDIST) \
		--search-directory $(DIAL) \
		--stdlib-directory $(WWW) \
		--filename stdlib.min.js

$(WWWDIST)/stdlib.full.js:
	- cp $(WWW)/brython_stdlib.js $(WWWDIST)/stdlib.full.js

$(WWWDIST)/index.html:
	- ln -s $(shell readlink -f $(WWW))/index.html $(WWWDIST)

$(WWWDIST)/test.html:
	- ln -s $(shell readlink -f $(WWW))/test.html $(WWWDIST)

$(WWWDIST)/webtests.py:
	- ln -s $(shell readlink -f $(WWW))/webtests.py $(WWWDIST)

.PHONY: www
www: $(WWWDIST)/stdlib.min.js $(WWWDIST)/dial.js $(WWWDIST)/index.html \
	$(WWWDIST)/test.html $(WWWDIST)/webtests.py
	- cp $(WWW)/brython.js $(WWWDIST)/runtime.js

.PHONY: serve
serve: www
	brython -C$(WWWDIST) serve

.PHONY: clear
clean::
	- rm \
		$(WWWDIST)/runtime.js \
		$(WWWDIST)/stdlib.*.js \
		$(WWWDIST)/dial.js \
		$(WWWDIST)/test.html \
		$(WWWDIST)/index.html
```
